fedimint-client/src/lib.rs

```diff
@@ -564,13 +564,26 @@ const SUPPORTED_CORE_API_VERSIONS: &[fedimint_core::module::ApiVersion] =
 pub type ModuleGlobalContextGen = ContextGen<DynGlobalClientContext>;
 
 /// Resources particular to a module instance
-pub struct ClientModuleInstance {
+pub struct ClientModuleInstance<'m, M: ClientModule> {
     /// Instance id of the module
     pub id: ModuleInstanceId,
     /// Module-specific DB
     pub db: Database,
     /// Module-specific API
     pub api: DynModuleApi,
```

Comment:
Separate issue: Should these be accessible via functions so we can change the way they are stored?

Reply:
Was thinking the same. Probably yes, so the owner can't assign at least?

Reply:
I could also imagine just having a normal dbtx reference in there and creating the module dbtx on demand etc.